spec/forms/locations_directory_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'rails_helper'
+
+RSpec.describe LocationsDirectory do
+  let(:user) { instance_double('User', pensionwise_admin?: true) }
+
+  describe 'display_active' do
+    it 'default to true when no value is set' do
+      expect(described_class.new(user, {}).display_active).to be_truthy
+    end
+
+    it 'false when value is an empty string' do
+      expect(described_class.new(user, locations_directory: { display_active: '' }).display_active).to be_falsey
+    end
+
+    it 'false when value is a`0`' do
+      expect(described_class.new(user, locations_directory: { display_active: '0' }).display_active).to be_falsey
+    end
+
+    it 'false when value is `false`' do
+      expect(described_class.new(user, locations_directory: { display_active: 'false' }).display_active).to be_falsey
+    end
+
+    it 'true for any other value' do
+      expect(described_class.new(user, locations_directory: { display_active: '1' }).display_active).to be_truthy
+      expect(described_class.new(user, locations_directory: { display_active: 'true' }).display_active).to be_truthy
+      expect(described_class.new(user, locations_directory: { display_active: 'apples' }).display_active).to be_truthy
+    end
+  end
+
+  describe 'display_hidden' do
+    it 'default to false when no value is set' do
+      expect(described_class.new(user, {}).display_hidden).to be_falsey
+    end
+
+    it 'false when value is an empty string' do
+      expect(described_class.new(user, locations_directory: { display_hidden: '' }).display_hidden).to be_falsey
+    end
+
+    it 'false when value is a`0`' do
+      expect(described_class.new(user, locations_directory: { display_hidden: '0' }).display_hidden).to be_falsey
+    end
+
+    it 'false when value is `false`' do
+      expect(described_class.new(user, locations_directory: { display_hidden: 'false' }).display_hidden).to be_falsey
+    end
+
+    it 'true for any other value' do
+      expect(described_class.new(user, locations_directory: { display_hidden: '1' }).display_hidden).to be_truthy
+      expect(described_class.new(user, locations_directory: { display_hidden: 'true' }).display_hidden).to be_truthy
+      expect(described_class.new(user, locations_directory: { display_hidden: 'apples' }).display_hidden).to be_truthy
+    end
+  end
+
+  describe '#locations' do
+    subject { described_class.new(user, params) }
+
+    context 'when hidden location exists' do
+      let!(:location) { FactoryGirl.create(:location, hidden: true) }
```

Comment:
Will do this in a separate PR as this project is currently setup to use `FactoryGirl.create`